src/parser/input-impl.h

```diff
@@ -257,6 +257,16 @@ inline bool ParseInput::takeSExprStart(std::string_view expected) {
   return false;
 }
 
+inline bool ParseInput::peekSExprStart(std::string_view expected) {
+  auto original = lexer;
+  if (!takeLParen()) {
+    return false;
+  }
+  bool ret = takeKeyword(expected);
+  lexer = original;
```

Comment:
How efficient is this?

Reply:
Very. It's essentially just resetting an index into the source buffer to its previous value.